### ISSUE TITLE
Invalid file signature error?

### ISSUE THREAD

**Issue by elsorino** [maharmstone/ntfs2btrfs#4]
What causes this error? I attempted to use this to convert a 2TB hdd but it gave me that error.
On arch linux if that matters

**Comment by lideming** [comment]
Same, also on Arch Linux, when I was trying to convert a Windows 10 OS drive.

Just searched in source and found the line that throws that error:
https://github.com/maharmstone/ntfs2btrfs/blob/171e9b92114ca08207afc34ccb190b4a9dfd90fb/src/ntfs.cpp#L75

**Comment by maharmstone** [comment]
Essentially it means "I've opened something that should be a file but isn't", which isn't very helpful. It looks like the program could do with being tested against more filesystems...

**Comment by isa-pp** [comment]
Same error on a 500gb hdd run on Ubuntu 20.04

**Comment by Inotamira** [comment]
Same here, actually, Manjaro. REALLY hoping to try this out on my "VR Only" Windows installation for giggles.

**Comment by YihaoPeng** [comment]
Same issue. GDB trace:

![截图录屏_选择区域_20200618020932](https://user-images.githubusercontent.com/26923506/85077611-4884fb80-b1f5-11ea-9d44-2eb7b72d7fda.png)

**Comment by yretenai** [comment]
Sector signature is 0, the entire block buffer is 0. Could be an empty block or a corrupt block? I have this error as well, CHKDSK says the disk is fine. Might be an issue relating to fragmented disks / SSDs and empty blocks.

**Comment by Inotamira** [comment]
> Sector signature is 0, the entire block buffer is 0. Could be an empty block or a corrupt block? I have this error as well, CHKDSK says the disk is fine. Might be an issue relating to fragmented disks / SSDs and empty blocks.
> 
> Program exiting without input is an issue relating to stderr flushing, should be fixed with my PR #5

Mine IS on an SSD now that you mention it

**Comment by TheGlitch76** [comment]
Reproduced on a 512GB NVME SSD.

**Comment by TheGlitch76** [comment]
I've looked into it further, and it appears that the underlying issue is that (as of Windows 10 build 2004) the Windows installer picks a cluster size of 512kb for partitions on an SSD.
The README states that Ntfs2Btrfs only supports cluster sizes of 4096kb.
It seems that for now a workaround would be to manually create the partition and give it a cluster size of 4096 when installing.

**Comment by timawesomeness** [comment]
There must be multiple underlying issues causing this error then, because I'm experiencing in on an HDD with 4096 cluster size.

**Comment by maharmstone** [comment]
Could you give it a go on the latest version please?

@TheGlitch76 - that's news to me, are you sure about that? There's nothing on https://support.microsoft.com/en-gb/help/140365/default-cluster-size-for-ntfs-fat-and-exfat, which says that Windows only chooses sizes other than 4KB on really massive disks. https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-fsa/4e3695bd-7574-4f24-a223-b4679c065b63 was updated a week ago, and nothing there either.

**Comment by Inotamira** [comment]
> Could you give it a go on the latest version please?
> 
> @TheGlitch76 - that's news to me, are you sure about that? There's nothing on https://support.microsoft.com/en-gb/help/140365/default-cluster-size-for-ntfs-fat-and-exfat, which says that Windows only chooses sizes other than 4KB on really massive disks. https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-fsa/4e3695bd-7574-4f24-a223-b4679c065b63 was updated a week ago, and nothing there either.

I now get Error seeking to ffffffffd781c000 when pointing it at my Windows partition.

**Comment by TheGlitch76** [comment]
> Could you give it a go on the latest version please?
> 
> @TheGlitch76 - that's news to me, are you sure about that? There's nothing on https://support.microsoft.com/en-gb/help/140365/default-cluster-size-for-ntfs-fat-and-exfat, which says that Windows only chooses sizes other than 4KB on really massive disks. https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-fsa/4e3695bd-7574-4f24-a223-b4679c065b63 was updated a week ago, and nothing there either.

hrm, I was probably measuring something other than what I thought (found it with linux commands).
Either way, it's getting past that now, but still failing.
Here's a [https://pastebin.com/wJVC5uZp](log), but I think the issue may be that the drive is nearly full. I'll have to do some reorganizing of my partitions and try again with more space.

**Comment by maharmstone** [comment]
Have you tried doing as it says - running chkdsk and trying again?

**Comment by TheGlitch76** [comment]
Yes, running chkdsk showed no errors, and a fresh install with sufficient space is giving the same issue.

**Comment by maharmstone** [comment]
Is this still an issue on today's version?

**Comment by alucryd** [comment]
> Is this still an issue on today's version?

Yes sir, on version 20210105: `Invalid file signature (00000000, expected 454c4946)`

**Comment by embedvr** [comment]
I am getting this issue on my 8TB hdd on Windows 10 (21H2)
`Invalid file signature (00000000, expected 454c4946).`

**Comment by r1sk01** [comment]
Still an issue on 20250616. (Might be regression if it was patched?)
```
❯ sudo ntfs2btrfs /dev/sda1
Using Zstd compression.
Using CRC32C for checksums.
Skipping encrypted inode 473c (\WpSystem\S-1-5-21-1704093480-2142582471-2815586854-1001\AppData\Local\Packages\Microsoft.SeaofThieves_8wekyb3d8bbwe\Settings\settings.dat.LOG1)
Invalid file signature (00000000, expected 454c4946).
```
The complaint about the encrypted inode happened way before the signature error.
